selfservice/hook/web_hook.go

```diff
@@ -173,12 +180,63 @@ func newWebHookConfig(r json.RawMessage) (*webHookConfig, error) {
 	}, nil
 }
 
-func NewWebHook(r webHookDependencies, c json.RawMessage) *WebHook {
-	return &WebHook{r: r, c: c}
+func NewWebHook(ctx context.Context, r webHookDependencies, c json.RawMessage) *WebHook {
+	tracer := r.Tracer(ctx)
+	client := createHttpClient(tracer)
+	return &WebHook{r: r, c: c, httpClient: client}
+}
+
+// The RoundTripperFunc type is an adapter to allow the use of ordinary
+// functions as RoundTrippers. If f is a function with the appropriate
+// signature, RountTripperFunc(f) is a RoundTripper that calls f.
+type RoundTripperFunc func(req *http.Request) (*http.Response, error)
+
+// RoundTrip implements the RoundTripper interface.
+func (rt RoundTripperFunc) RoundTrip(r *http.Request) (*http.Response, error) {
+	return rt(r)
+}
+
+func createHttpClient(tracer *tracing.Tracer) *http.Client {
```

Comment:
Is there some way to write a test for this functionality? It looks fine but we need to ensure:

1. It works
2. It doesn't break when someone changes something